neuraxle/pipeline.py

```diff
@@ -369,99 +277,56 @@ class ResumablePipeline(Pipeline, ResumableStepMixin):
-    def save(self, data_container: DataContainer):
-        """
-        Save the fitted parent pipeline with the passed data container
-
-        :param data_container: data container to save pipeline with
-        :return:
-        """
-        if self.parent_step is None:
-            self.pipeline_saver.save(self, data_container)
-        else:
-            self.parent_step.save(data_container)
-
-    def _load_checkpoint(self, data_container: DataContainer) -> Tuple[NamedTupleList, DataContainer]:
+    def _load_checkpoint(
+            self,
+            data_container: DataContainer,
+            context: ExecutionContext
+    ) -> Tuple[NamedTupleList, DataContainer]:
         """
         Try loading a pipeline cache with the passed data container.
         If pipeline cache loading succeeds, find steps left to do,
         and load the latest data container.
 
         :param data_container: the data container to resume
+        :param context: the execution context to resume
         :return: tuple(steps left to do, last checkpoint data container)
         """
         new_starting_step_index, starting_step_data_container = \
-            self._get_starting_step_info(data_container)
-
-        if self.parent_step is None:
-            if not self.pipeline_saver.can_load(self, starting_step_data_container):
-                return self.steps_as_tuple, data_container
+            self._get_starting_step_info(data_container, context)
 
-            saved_pipeline = self.pipeline_saver.load(self, starting_step_data_container)
+        loaded_pipeline = self.load(context)
+        if not self.are_steps_before_index_the_same(loaded_pipeline, new_starting_step_index):
+            return self.steps_as_tuple, data_container
 
-            if not self._can_load_saved_pipeline(
-                    saved_pipeline=saved_pipeline,
-                    starting_step_data_container=starting_step_data_container,
-                    new_starting_step_index=new_starting_step_index
-            ):
-                return self.steps_as_tuple, data_container
-
-            self._load_saved_pipeline_steps_before_index(
-                saved_pipeline=saved_pipeline,
-                index=new_starting_step_index
-            )
+        self._load_pipeline(loaded_pipeline)
 
         step = self[new_starting_step_index]
         if isinstance(step, BaseCheckpointStep):
             starting_step_data_container = step.read_checkpoint(starting_step_data_container)
 
         return self[new_starting_step_index:], starting_step_data_container
 
-    def _can_load_saved_pipeline(
-            self,
-            saved_pipeline: Pipeline,
-            starting_step_data_container: DataContainer,
-            new_starting_step_index
-    ) -> bool:
-        """
-        Returns True if the saved pipeline steps before passed starting step index
-        are the same as current pipeline steps before starting step index.
+    def _load_pipeline(self, loaded_self):
```

Comment:
Why is this method here tho?

Reply:
after calling self.load(context), we can load the loaded pipeline into self... renaming to load_loaded_pipeline_into_self, and resolving